src/server/dev.ts

```diff
@@ -73,7 +76,21 @@ function init() {
         { frequency: 60, simulator: ChartSimulator.of() },
       ],
     })
-    virtualRobots.startSimulators()
+    let stopSimulators: () => void | undefined
+    reaction(
+      () => webSocketProxyNUClearNetServer.numConnections > 0,
+      enable => {
+        if (enable) {
+          // tslint:disable-next-line no-console
+          console.log('Client connected, starting simulator')
+          stopSimulators = virtualRobots.startSimulators()
+        } else if (stopSimulators) {
+          // tslint:disable-next-line no-console
+          console.log('No clients remain connected, stopping simulator')
+        }
+      },
+      { fireImmediately: true },
+    )
```

Comment:
Uglier than needed I suspect. Perhaps `VirtualRobots` should instead expose a more flexible API that would be more suitable for reacting to an observable value.

Something like:

```ts
reaction(() => {
  () => now(frequency) && numConections > 0,
  enabled => enabled && virtualRobots.tick()
})
```

Will try this later.